kubernetes/schema_pod_spec.go

```diff
@@ -254,6 +254,12 @@ func podSpecFields(isUpdatable, isDeprecated, isComputed bool) map[string]*schem
 			Description: "ServiceAccountName is the name of the ServiceAccount to use to run this pod. More info: http://releases.k8s.io/HEAD/docs/design/service_accounts.md.",
 			Deprecated:  deprecatedMessage,
 		},
+		"share_process_namespace": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Default:     false,
+			Description: "When process namespace sharing is enabled, processes in a container are visible to all other containers in that pod"
```

Comment:
I'd reuse the complete description from the [reference documentation](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.15/#podspec-v1-core):

```suggestion
			Description: "Share a single process namespace between all of the containers in a pod. When this is set containers will be able to view and signal processes from other containers in the same pod, and the first process in each container will not be assigned PID 1. HostPID and ShareProcessNamespace cannot both be set. Optional: Defaults to false."	
```

Also, this should probably declared as `ConflictsWith: ["host_pid"]`

Reply:
> Also, this new attribute should probably declared as ConflictsWith: ["host_pid"].

 Sorry, can you explain it?

Reply:
>  acceptance tests should be added to ensure this can be enabled then disabled
Should it be like a `TestAccKubernetesDeployment_basic`?

Reply:
> > Also, this new attribute should probably declared as ConflictsWith: ["host_pid"].
> 
> Sorry, can you explain it?

The doc says :
> HostPID and ShareProcessNamespace cannot both be set.

So I would probably have added `ConflictsWith: []string{"host_pid"}` here.

However, I just realized this is more tricky than I thought as this part of the schema is shared by many resources (Pod, Deployment, DaemonSet, etc.) and the names specified in `ConflictsWith` must be absolute, i.e. `spec.0.template.0.spec.0.host_pid` in case for workload managers and `spec.0.host_pid` for pods.

I'd say forget about that for now.

Reply:
> Should it be like a `TestAccKubernetesDeployment_basic`?

Precisely :+1:

Reply:
> I'd say forget about that for now.

Oh, ok. It's really difficult